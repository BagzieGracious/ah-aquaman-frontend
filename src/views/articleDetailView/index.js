import React, { Component } from "react";
import { connect } from "react-redux";
import { ToastContainer, toast } from "react-toastify";
import ArticleComponent from "../../components/article/articleComponent";
import getSingleArticle from "../../actions/articleActions/getSingleArticle";
import CircularProgressLoader from "../../commons/progressLoader";
import Sidebar from "../../components/article/sidebar";
import Navbar from "../../components/article/navbarComponent/Navbar";
import ImageBanner from "../../components/article/ImageBanner";
import { bookmarkArticleAction, bookmarkListing } from "../../actions/articleActions/bookmarkAction";
import { getComments } from "../../actions/commentActions/index";

function ShareLinks(article, slug) {
  const articleLink = `${process.env.FRONTEND_BASE_URL}/article/${slug}`;
  const shareBody = `I found this article to be worth reading, ${article.title} ${articleLink}`;
  const mailshare = `mailto:?Subject=${encodeURI(article.title)}&body=${shareBody}`;
  const facebookShare = `https://www.facebook.com/sharer/sharer.php?u=${encodeURI(articleLink)}`;
  const twitterShare = `https://twitter.com/home?status=${encodeURI(shareBody)}`;

  return {
    mailshare, facebookShare, twitterShare,
  };
}

export class ArticleDetailView extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: true,
    };
  }

  componentWillMount() {
    const { props } = this;
    const { slug } = props.match.params;

    props.getSingleArticle(slug);
    localStorage.setItem("slug", slug);
    props.bookmarkListing(slug);
    props.getComments(slug);
  }

  componentWillReceiveProps(props) {
    this.setState({ loading: props.article.loading });
  }

  handleBookmark = (e) => {
    e.preventDefault();
    const { slug } = this.props.match.params;
    const { isBookmarked } = this.props;
    this.props.bookmarkArticleAction(slug, isBookmarked);
  }

  render() {   
    const { article } = this.props;
    const loader = this.state;
    const loggedInUsername = localStorage.getItem("username");
    const articleAuthorUsername = article.author.username;
    if (this.state.loading) {
      return (
        <CircularProgressLoader {...loader} />
      );
    }

    if (article.title === "") {
      return (
        <div className="h-100 w-100 p-5 m-5  text-center">
          {toast.error("Article not found", {
            position: toast.POSITION.TOP_CENTER,
            autoClose: 3000,
            hideProgressBar: false,
            pauseOnHover: true,
          })}
;
        </div>
      );
    }
    article.shareLinks = ShareLinks(article, this.props.match.params.slug);
    return (

      <React.Fragment>
        <Navbar />
        <ImageBanner info="View Article" page="view_article" />
        <section className="blog-section">
          <div className="container">
            <div className="row">
              <ArticleComponent
                article={article}
                slug={this.props.match.params.slug}
                handleBookmark={this.handleBookmark}
                isBookmarked={this.props.isBookmarked}
                loggedInUsername={loggedInUsername}
                articleAuthorUsername={articleAuthorUsername}
              />
              <Sidebar />
            </div>
            <ToastContainer />
          </div>
        </section>
      </React.Fragment>
    );
  }
}

export const mapStateToProps = state => ({
  loading: state.ArticleReducer.loading,
  article: state.ArticleReducer.article,
  isBookmarked: state.bookmarkReducer.isBookmarked,
  message: state.bookmarkReducer.message,
});

export default connect(mapStateToProps, 
  { getSingleArticle, bookmarkArticleAction, bookmarkListing, getComments })(ArticleDetailView);
