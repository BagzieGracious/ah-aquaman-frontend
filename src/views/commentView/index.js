import React, { Component } from "react";
import { connect } from "react-redux";
import Moment from "react-moment";
import ReactTimeAgo from "react-time-ago";
import {addComment} from "../../actions/commentActions/index";

export class CommentView extends Component {
  handleSubmit = (e) => {
    e.preventDefault();
    const comment = (e.target.elements.comment.value).trim();
    const slugg = (e.target.elements.slugg.value);
    if(comment){
        this.props.addComment( slugg, comment );
    }
  };

  render() {
    const { comments } = this.props;
    const { slug }=this.props;
    
    return (
        <React.Fragment>
            <h3>{(comments).length} Comments </h3>
            <div className="commentslist">
                {
                    comments.length > 0 
                    ? comments.map((comment) => (
                        <div key={comment.id} className="comment">
                            <div className="media">
                                <div className="media-body">
                                    <h4>
                                        <label className="float-left">{comment.body}</label>
                                        <small className="float-right">{<Moment fromNow>{comment.created_at}</Moment>}</small>
                                    </h4>
                                    <p>{comment.commented_by.username}</p>
                                    <div className="separator" />
                                </div>    
                            </div>    
                        </div>
                    ))
                    :""
                }
            </div>
            <form onSubmit={this.handleSubmit}>
                <div className="form-group">
                    <input  type="hidden" name="slugg" className="form-control mt-3" value={slug} />
                    <textarea name="comment" className="form-control" placeholder="Enter your comment" />
                    <button type="submit" onClick={(e)=>{
                        {this.isLoggedIn()?document.location.href="/login" : (this.handleSubmit);
                    }}} className="btn btn-default">{this.isLoggedIn()?"Login to comment":"Post Comment"}</button>
                </div>
            </form>
        </React.Fragment>
    );
  }

isLoggedIn=()=>localStorage.getItem("token")===null;
}

export const mapStateToProps = state => ({
  comments: state.commentReducer.comments,
});


export default connect(mapStateToProps, { addComment })(CommentView);
