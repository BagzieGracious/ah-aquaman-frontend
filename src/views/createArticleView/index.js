import React, { Component } from "react";
import { connect } from "react-redux";
import { ToastContainer, toast } from "react-toastify";
import Editor from "../../components/article/articleEditor";
import { CreateArticle } from "../../actions/articleActions/createArticle";
import Sidebar from "../../components/article/sidebar";
import Navbar from "../../components/article/navbarComponent/Navbar";
import ImageBanner from "../../components/article/ImageBanner";

export class CreateArticleView extends Component {
  constructor() {
    super();
    this.state = {
      title: "",
      description: "",
      body: "",
      tagList: [],
    };

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleInput = this.handleInput.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (!nextProps.isSuccessful) {
      this.handleErrors(nextProps);
    } else {
      this.props.history.push(`/article/${nextProps.article.slug}`);
    }
  }

  handleChange=(tagList) => {
    const sorted = [];
    tagList.forEach((element) => {
      sorted.push(element.toLowerCase());
    });
    sorted.sort();
    this.setState({
      tagList: Array.from(new Set(sorted)),
    });
  }

  handleErrors = (props) => {
    if (props.errors.title) {
      toast.error(props.errors.title);
    }
    if (props.errors.description) {
      toast.error(props.errors.description);
    }
    if (props.errors.body) {
      toast.error(props.errors.body);
    }
  }

  handleSubmit(event) {
    event.preventDefault();
    const payload = {
      title: this.state.title,
      description: this.state.description,
      body: this.state.body,
      tagList: this.state.tagList,
    };

    this.props.CreateArticle(payload);
  }

  handleInput(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  render() {
    return (
      <React.Fragment>
        <Navbar />
        <ImageBanner info="Create Article" page="create_article" />
        <div className="container">
          <div className="row">
            <Editor

              {...this.state}
              onSubmit={this.handleSubmit}
              onInputChange={this.handleInput}
              value={this.state.tagList}
              onChange={this.handleChange}
            />

            <Sidebar />
          </div>
          <ToastContainer />
        </div>
      </React.Fragment>
    );
  }
}


export const mapStateToProps = state => ({
  isSuccessful: state.articleCreateReducer.isSuccessful,
  errors: state.articleCreateReducer.errors,
  article: state.articleCreateReducer.article,
});

export default connect(mapStateToProps, { CreateArticle })(CreateArticleView);
