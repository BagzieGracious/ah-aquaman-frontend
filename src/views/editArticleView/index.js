import React, { Component } from "react";
import { connect } from "react-redux";
import { ToastContainer, toast } from "react-toastify";
import Editor from "../../components/article/articleEditor";
import { EditArticle } from "../../actions/articleActions/createArticle";
import getSingleArticle from "../../actions/articleActions/getSingleArticle";
import Sidebar from "../../components/article/sidebar";
import Navbar from "../../components/article/navbarComponent/Navbar";
import ImageBanner from "../../components/article/ImageBanner";

export class EditArticleView extends Component {
  constructor() {
    const retrivedArticle = JSON.parse(localStorage.getItem("article"));
    super();
    this.state = {
      title: retrivedArticle.title,
      description: retrivedArticle.description,
      body: retrivedArticle.body,
      tags: retrivedArticle.tagList,
    };

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleInput = this.handleInput.bind(this);
  }

  componentDidMount() {
    if (this.props.match.params.slug) {
      this.props.getSingleArticle(this.props.match.params.slug);
    }
  }

  componentWillReceiveProps(nextProps) {
    if (!nextProps.isEditSuccessful) {
      this.handleErrors(nextProps);
    }

    if (nextProps.isEditSuccessful) {
      this.props.history.push(`/article/${nextProps.match.params.slug}`);
    }
  }

  handleErrors = (props) => {
    if (props.errors.title) {
      toast.error("title can not be blank");
    }

    if (props.errors.description) {
      toast.error("description can not be blank");
    }

    if (props.errors.body) {
      toast.error("Body can not be blank");
    }
  }

  handleSubmit(event) {
    event.preventDefault();
    const editPayload = {
      title: this.state.title,
      description: this.state.description,
      body: this.state.body,
    };
    this.props.EditArticle(this.props.match.params.slug, editPayload);
  }

  handleInput(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  render() {
    return (
      <React.Fragment>
        <Navbar />
        <ImageBanner info="Edit Article" page="editor" />
        <div className="container">
          <div className="row">
            <Editor
              {...this.state}
              onSubmit={this.handleSubmit}
              onInputChange={this.handleInput}
              slug={this.props.slug}
            />
            <Sidebar />
          </div>
          <ToastContainer />
        </div>
      </React.Fragment>
      
    );
  }
}

export const mapStateToProps = state => ({
  errors: state.articleCreateReducer.errors,
  isEditSuccessful: state.articleCreateReducer.isEditSuccessful,
});

export default connect(mapStateToProps, { getSingleArticle, EditArticle })(EditArticleView);
