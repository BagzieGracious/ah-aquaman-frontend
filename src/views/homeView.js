import React from "react";
import Navbar from "../components/article/navbarComponent/Navbar";
import ImageBanner from "../components/article/ImageBanner";
import Articles from "../components/article/articleHomeComponent";
import Sidebar from "../components/article/sidebar";

const HomeView = () => (
  <React.Fragment>
    <Navbar />
    { location.pathname !== "/" && <ImageBanner info="Home Page" page="Home" /> }
    <section className="blog-section">
      <div className="container">
        <div className="row">
          <Articles />
          <Sidebar />
        </div>
      </div>
    </section>
  </React.Fragment>
);

export default HomeView;
