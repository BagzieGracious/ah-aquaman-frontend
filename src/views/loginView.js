import React from "react";
import Login from "../components/auth/loginComponent";

const LoginView = () => (
  <div>
    <Login />
  </div>
);

export default LoginView;
