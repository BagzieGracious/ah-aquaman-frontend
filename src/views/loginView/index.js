import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { toast } from "react-toastify";
import Login from "../../components/loginForm";
import loginAction from "../../actions/loginAction";
import Authenticate from "../../routes/protectedRoutes";
import CircularProgressLoader from "../../commons/progressLoader";

const jwt = require("jsonwebtoken");

export class LoginView extends Component {
  state = {
    loader: {
      success: false,
      loading: false,
    },
    isProcessing: false,
  };

  componentDidMount() {
    if (Authenticate(localStorage.getItem("token"))) {
      this.props.history.push("/");
    }
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.token) {
      localStorage.setItem("token", nextProps.token);
      
      const decoded = jwt.decode(nextProps.token);
      if (decoded) {
        const username = decoded.username;
        localStorage.setItem("username", username);
        this.props.history.push("/");
        toast.success("You have successfully signed in", {
          position: toast.POSITION.TOP_CENTER,
          autoClose: 3000,
          hideProgressBar: false,
          pauseOnHover: true,
        });
      }
    }
    this.setState({ loader: { loading: false } });
    this.setState({ isProcessing: false });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const { loginAction } = this.props;
    const userData = {
      email: event.target.elements.email.value,
      password: event.target.elements.password.value,
    };
    loginAction(userData);
    this.setState({ loader: { loading: true } });
    this.setState({ isProcessing: true });
  };

  onChangeRegister = () => {
    this.props.history.push("/signup");
  }

  onChangeResetPassword = () => {
    this.props.history.push("/reset-password");
  }

  render() {
    const { errors, history } = this.props;
    const loader = this.state.loader;
    const buttonDisable = this.state.isProcessing;
    return (
      <div className="login background-image">
        <CircularProgressLoader {...loader} />
        <Login
          onSubmit={this.handleSubmit}
          isProcessing={buttonDisable}
          errors={errors}
          history={history}
          onRegister={this.onChangeRegister}
          onPassword={this.onChangeResetPassword}
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  token: state.loginReducer.token,
  errors: state.loginReducer.errors,
  isSuccessful: state.loginReducer.isSuccessful,
});

LoginView.propTypes = {
  token: PropTypes.string,
  errors: PropTypes.shape({
    email: PropTypes.any,
    password: PropTypes.any,
    error: PropTypes.any,
  }),
};

LoginView.defaultProps = {
  token: "",
  errors: {},
};

export default connect(mapStateToProps, { loginAction })(LoginView);
