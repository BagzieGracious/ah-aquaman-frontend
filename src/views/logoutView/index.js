import React, { Component } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router";
import { toast, ToastContainer } from "react-toastify";

export class LogoutPage extends Component {
  componentWillMount() {
    localStorage.removeItem("token");
    localStorage.removeItem("username");
    localStorage.removeItem("article");
    localStorage.removeItem("slug");
    localStorage.removeItem("image");
    this.props.history.push("/");
    toast.success("You have successfully Logged out", {
      position: toast.POSITION.TOP_CENTER,
      autoClose: 3000,
      hideProgressBar: false,
      pauseOnHover: true,
    });
  }

  render() {
    return (
      <div>
        <ToastContainer />
      </div>
    );
  }
}

export default LogoutPage;