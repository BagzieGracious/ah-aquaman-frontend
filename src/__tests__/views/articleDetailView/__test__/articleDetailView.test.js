import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow, mount } from "enzyme";
import { ArticleDetailView, mapStateToProps } from "../../../../views/articleDetailView";


describe("ArticleDetailView", () => {
  it("should render without crashing", () => {
    const props = {
      article: {
        title: "test title",
        author: "testuser",
      },
      loading: false,
      match: {
        params: {
          slug: "article-slug",
        },
      },
      getSingleArticle: jest.fn(),
      getComments: jest.fn(),
    };

    const instance = shallow(<ArticleDetailView {...props} />);
    instance.setProps({
      loading: false,
    });
    expect(instance).toMatchSnapshot();
  });

  it("should render 404 view without crashing", () => {
    const props = {
      article: {
        title: "",
      },
      loading: true,
      match: {
        params: {
          slug: "article-slug",
        },
      },
      getSingleArticle: jest.fn(),
    };
    const wrapper = new ArticleDetailView(props);
    wrapper.state = { notFound: true };
    expect(wrapper.render()).toMatchSnapshot();
  });


  it("should call getSingleArticle after will mount ", () => {
    const props = {
      articleReducer: {
        article: {
          title: "test title",
        },
      },
      match: {
        params: {
          slug: "article-slug",
        },
      },
      getSingleArticle: jest.fn(),
      getComments: jest.fn(),
    };
    const wrapper = mount(<Router><ArticleDetailView {...props} /></Router>);
    const spyProp = jest.spyOn(wrapper.instance().props.children.props, "getSingleArticle");
    expect(spyProp).toHaveBeenCalled();
  });

  it("should map state to props", () => {
    const state = {
      ArticleReducer: {
        article: { title: "test article", description: "test article", body: "test article" },
        loading: true,
      },
    };
    expect(mapStateToProps(state)).toEqual({
      article: { title: "test article", description: "test article", body: "test article" },
      loading: true,
    });
  });
});
