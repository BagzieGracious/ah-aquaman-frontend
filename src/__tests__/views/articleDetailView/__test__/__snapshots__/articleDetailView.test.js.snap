// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleDetailView should render 404 view without crashing 1`] = `
<div
  className="h-100 w-100 p-5 m-5  text-center"
>
  fi5spgrd2p
  ;
</div>
`;

exports[`ArticleDetailView should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleDetailView
    article={
      Object {
        "author": "testuser",
        "title": "test title",
      }
    }
    getSingleArticle={
      [MockFunction] {
        "calls": Array [
          Array [
            "article-slug",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    loading={false}
    match={
      Object {
        "params": Object {
          "slug": "article-slug",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(Navbar) />,
        <ImageBanner
          info="View Article"
          page="view_article"
        />,
        <section
          className="blog-section"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <ArticleComponent
                article={
                  Object {
                    "author": "testuser",
                    "title": "test title",
                  }
                }
                slug="article-slug"
              />
              <Sidebar />
            </div>
            <ToastContainer
              autoClose={5000}
              bodyClassName={null}
              className={null}
              closeButton={
                <CloseButton
                  ariaLabel="close"
                />
              }
              closeOnClick={true}
              draggable={true}
              draggablePercent={80}
              hideProgressBar={false}
              newestOnTop={false}
              pauseOnFocusLoss={true}
              pauseOnHover={true}
              position="top-right"
              progressClassName={null}
              progressStyle={null}
              rtl={false}
              style={null}
              toastClassName={null}
              transition={[Function]}
            />
          </div>
        </section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "info": "View Article",
          "page": "view_article",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <div
              className="row"
            >
              <ArticleComponent
                article={
                  Object {
                    "author": "testuser",
                    "title": "test title",
                  }
                }
                slug="article-slug"
              />
              <Sidebar />
            </div>
            <ToastContainer
              autoClose={5000}
              bodyClassName={null}
              className={null}
              closeButton={
                <CloseButton
                  ariaLabel="close"
                />
              }
              closeOnClick={true}
              draggable={true}
              draggablePercent={80}
              hideProgressBar={false}
              newestOnTop={false}
              pauseOnFocusLoss={true}
              pauseOnHover={true}
              position="top-right"
              progressClassName={null}
              progressStyle={null}
              rtl={false}
              style={null}
              toastClassName={null}
              transition={[Function]}
            />
          </div>,
          "className": "blog-section",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="row"
              >
                <ArticleComponent
                  article={
                    Object {
                      "author": "testuser",
                      "title": "test title",
                    }
                  }
                  slug="article-slug"
                />
                <Sidebar />
              </div>,
              <ToastContainer
                autoClose={5000}
                bodyClassName={null}
                className={null}
                closeButton={
                  <CloseButton
                    ariaLabel="close"
                  />
                }
                closeOnClick={true}
                draggable={true}
                draggablePercent={80}
                hideProgressBar={false}
                newestOnTop={false}
                pauseOnFocusLoss={true}
                pauseOnHover={true}
                position="top-right"
                progressClassName={null}
                progressStyle={null}
                rtl={false}
                style={null}
                toastClassName={null}
                transition={[Function]}
              />,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <ArticleComponent
                    article={
                      Object {
                        "author": "testuser",
                        "title": "test title",
                      }
                    }
                    slug="article-slug"
                  />,
                  <Sidebar />,
                ],
                "className": "row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "article": Object {
                      "author": "testuser",
                      "title": "test title",
                    },
                    "slug": "article-slug",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoClose": 5000,
                "bodyClassName": null,
                "className": null,
                "closeButton": <CloseButton
                  ariaLabel="close"
                />,
                "closeOnClick": true,
                "draggable": true,
                "draggablePercent": 80,
                "hideProgressBar": false,
                "newestOnTop": false,
                "pauseOnFocusLoss": true,
                "pauseOnHover": true,
                "position": "top-right",
                "progressClassName": null,
                "progressStyle": null,
                "rtl": false,
                "style": null,
                "toastClassName": null,
                "transition": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "section",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(Navbar) />,
          <ImageBanner
            info="View Article"
            page="view_article"
          />,
          <section
            className="blog-section"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <ArticleComponent
                  article={
                    Object {
                      "author": "testuser",
                      "title": "test title",
                    }
                  }
                  slug="article-slug"
                />
                <Sidebar />
              </div>
              <ToastContainer
                autoClose={5000}
                bodyClassName={null}
                className={null}
                closeButton={
                  <CloseButton
                    ariaLabel="close"
                  />
                }
                closeOnClick={true}
                draggable={true}
                draggablePercent={80}
                hideProgressBar={false}
                newestOnTop={false}
                pauseOnFocusLoss={true}
                pauseOnHover={true}
                position="top-right"
                progressClassName={null}
                progressStyle={null}
                rtl={false}
                style={null}
                toastClassName={null}
                transition={[Function]}
              />
            </div>
          </section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "info": "View Article",
            "page": "view_article",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
            >
              <div
                className="row"
              >
                <ArticleComponent
                  article={
                    Object {
                      "author": "testuser",
                      "title": "test title",
                    }
                  }
                  slug="article-slug"
                />
                <Sidebar />
              </div>
              <ToastContainer
                autoClose={5000}
                bodyClassName={null}
                className={null}
                closeButton={
                  <CloseButton
                    ariaLabel="close"
                  />
                }
                closeOnClick={true}
                draggable={true}
                draggablePercent={80}
                hideProgressBar={false}
                newestOnTop={false}
                pauseOnFocusLoss={true}
                pauseOnHover={true}
                position="top-right"
                progressClassName={null}
                progressStyle={null}
                rtl={false}
                style={null}
                toastClassName={null}
                transition={[Function]}
              />
            </div>,
            "className": "blog-section",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="row"
                >
                  <ArticleComponent
                    article={
                      Object {
                        "author": "testuser",
                        "title": "test title",
                      }
                    }
                    slug="article-slug"
                  />
                  <Sidebar />
                </div>,
                <ToastContainer
                  autoClose={5000}
                  bodyClassName={null}
                  className={null}
                  closeButton={
                    <CloseButton
                      ariaLabel="close"
                    />
                  }
                  closeOnClick={true}
                  draggable={true}
                  draggablePercent={80}
                  hideProgressBar={false}
                  newestOnTop={false}
                  pauseOnFocusLoss={true}
                  pauseOnHover={true}
                  position="top-right"
                  progressClassName={null}
                  progressStyle={null}
                  rtl={false}
                  style={null}
                  toastClassName={null}
                  transition={[Function]}
                />,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <ArticleComponent
                      article={
                        Object {
                          "author": "testuser",
                          "title": "test title",
                        }
                      }
                      slug="article-slug"
                    />,
                    <Sidebar />,
                  ],
                  "className": "row",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "article": Object {
                        "author": "testuser",
                        "title": "test title",
                      },
                      "slug": "article-slug",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "autoClose": 5000,
                  "bodyClassName": null,
                  "className": null,
                  "closeButton": <CloseButton
                    ariaLabel="close"
                  />,
                  "closeOnClick": true,
                  "draggable": true,
                  "draggablePercent": 80,
                  "hideProgressBar": false,
                  "newestOnTop": false,
                  "pauseOnFocusLoss": true,
                  "pauseOnHover": true,
                  "position": "top-right",
                  "progressClassName": null,
                  "progressStyle": null,
                  "rtl": false,
                  "style": null,
                  "toastClassName": null,
                  "transition": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "section",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
